File: cleanup_workspace.js
const fs = require('fs');
const path = require('path');

const baseDir = "c:\\Users\\User\\Raizys-site";

const filesToDelete = [
    "add-topbar.ps1",
    "byte-clean.ps1",
    "check_active.js",
    "check_all_tags.js",
    "check_boms.js",
    "check_cal_f_rules.js",
    "check_css_encoding.js",
    "check_css_sections.js",
    "check_fixes.ps1",
    "check_home_sections.js",
    "check_null_bytes.js",
    "check_solucoes.js",
    "check_tags.js",
    "clean-artifacts.ps1",
    "clean_artifacts.py",
    "close.js",
    "correct_words.txt",
    "bad_words.txt",
    "css_additions.css",
    "diff.txt",
    "final-clean.ps1",
    "final-surgical-clean.ps1",
    "find_bad_words.ps1",
    "find_flying_cal_styles.js",
    "find_footer_styles.js",
    "fix_all.js",
    "fix_chars.ps1",
    "fix_encoding.js",
    "fix_final.py",
    "fix_forms.js",
    "fix_home.js",
    "fix_mobile_cache.js",
    "git.ignore",
    "inline_css.js",
    "migrate_web3forms.js",
    "precise-clean.ps1",
    "read_agenda_css.js",
    "read_all_changes.js",
    "read_css_diff.js",
    "read_css_sections.js",
    "read_default_eco_bar.js",
    "read_diff.js",
    "read_eco_bar_css.js",
    "read_header_diff.js",
    "read_header_selectors.js",
    "read_home_changes.js",
    "reverse_double_encoding.py",
    "standardize-and-clean.ps1",
    "surgical-clean.ps1",
    "test.txt",
    "update-footer-address.ps1",
    "update-header.ps1",
    "update-pages.ps1",
    "update-version.ps1",
    "update_footer.py",
    ".FullName",
    "assets/1.png",
    "assets/2.png",
    "assets/header-reference.png",
    "assets/hero-reference.png",
    "assets/logo-icon.png",
    "assets/logo-text.png",
    "assets/partner.png",
    "assets/pet_hero_illustration.png"
];

const dirsToDelete = [
    "backup_temp_2",
    "backup_temp_clean",
    "temp_zip"
];

function deleteUnnecessary() {
    console.log("Deleting unnecessary files...");
    for (const f of filesToDelete) {
        const filepath = path.join(baseDir, f);
        if (fs.existsSync(filepath)) {
            try {
                fs.unlinkSync(filepath);
                console.log(`Deleted file: ${f}`);
            } catch (err) {
                console.error(`Error deleting file ${f}:`, err.message);
            }
        }
    }

    console.log("Deleting unnecessary directories...");
    for (const d of dirsToDelete) {
        const dirpath = path.join(baseDir, d);
        if (fs.existsSync(dirpath) && fs.statSync(dirpath).isDirectory()) {
            try {
                fs.rmSync(dirpath, { recursive: true, force: true });
                console.log(`Deleted directory: ${d}`);
            } catch (err) {
                console.error(`Error deleting directory ${d}:`, err.message);
            }
        }
    }
}

function stripCssComments(text) {
    let out = [];
    let i = 0;
    let n = text.length;
    let inComment = false;
    let inStringSingle = false;
    let inStringDouble = false;

    while (i < n) {
        let c = text[i];
        let nextC = i + 1 < n ? text[i + 1] : "";

        if (!inComment) {
            if (!inStringSingle && !inStringDouble) {
                if (c === '/' && nextC === '*') {
                    inComment = true;
                    i++;
                } else if (c === "'") {
                    inStringSingle = true;
                    out.push(c);
                } else if (c === '"') {
                    inStringDouble = true;
                    out.push(c);
                } else {
                    out.push(c);
                }
            } else if (inStringSingle) {
                out.push(c);
                if (c === '\\') {
                    out.push(nextC);
                    i++;
                } else if (c === "'") {
                    inStringSingle = false;
                }
            } else if (inStringDouble) {
                out.push(c);
                if (c === '\\') {
                    out.push(nextC);
                    i++;
                } else if (c === '"') {
                    inStringDouble = false;
                }
            }
        } else {
            if (c === '*' && nextC === '/') {
                inComment = false;
                i++;
            }
        }
        i++;
    }
    return out.join("");
}

function stripJsComments(text) {
    let out = [];
    let i = 0;
    let n = text.length;
    let state = "NORMAL";

    function getLastNonWs() {
        for (let j = out.length - 1; j >= 0; j--) {
            if (!/\s/.test(out[j])) {
                return out[j];
            }
        }
        return "";
    }

    while (i < n) {
        let c = text[i];
        let nextC = i + 1 < n ? text[i + 1] : "";

        if (state === "NORMAL") {
            if (c === '"') {
                state = "STR_DOUBLE";
                out.push(c);
            } else if (c === "'") {
                state = "STR_SINGLE";
                out.push(c);
            } else if (c === '`') {
                state = "STR_TEMPLATE";
                out.push(c);
            } else if (c === '/' && nextC === '/') {
                state = "LINE_COMMENT";
                i++;
            } else if (c === '/' && nextC === '*') {
                state = "BLOCK_COMMENT";
                i++;
            } else if (c === '/') {
                let lastChar = getLastNonWs();
                if (lastChar && (/[a-zA-Z0-9_$]/.test(lastChar) || lastChar === ')' || lastChar === ']' || lastChar === '}')) {
                    out.push(c);
                } else {
                    state = "REGEX";
                    out.push(c);
                }
            } else {
                out.push(c);
            }
        } else if (state === "STR_DOUBLE") {
            out.push(c);
            if (c === '\\') {
                out.push(nextC);
                i++;
            } else if (c === '"') {
                state = "NORMAL";
            }
        } else if (state === "STR_SINGLE") {
            out.push(c);
            if (c === '\\') {
                out.push(nextC);
                i++;
            } else if (c === "'") {
                state = "NORMAL";
            }
        } else if (state === "STR_TEMPLATE") {
            out.push(c);
            if (c === '\\') {
                out.push(nextC);
                i++;
            } else if (c === '`') {
                state = "NORMAL";
            }
        } else if (state === "REGEX") {
            out.push(c);
            if (c === '\\') {
                out.push(nextC);
                i++;
            } else if (c === '[') {
                state = "REGEX_CLASS";
            } else if (c === '/') {
                state = "NORMAL";
            }
        } else if (state === "REGEX_CLASS") {
            out.push(c);
            if (c === '\\') {
                out.push(nextC);
                i++;
            } else if (c === ']') {
                state = "REGEX";
            }
        } else if (state === "LINE_COMMENT") {
            if (c === '\n') {
                state = "NORMAL";
                out.push(c);
            }
        } else if (state === "BLOCK_COMMENT") {
            if (c === '*' && nextC === '/') {
                state = "NORMAL";
                i++;
            }
        }
        i++;
    }
    return out.join("");
}

function stripHtmlComments(text) {
    let out = [];
    let i = 0;
    let n = text.length;

    while (i < n) {
        if (text.substring(i, i + 4) === '<!--') {
            i += 4;
            while (i < n && text.substring(i, i + 3) !== '-->') {
                i++;
            }
            i += 3;
            continue;
        }

        if (text.substring(i, i + 6).toLowerCase() === '<style') {
            let startTagEnd = text.indexOf('>', i);
            if (startTagEnd !== -1) {
                out.push(text.substring(i, startTagEnd + 1));
                let closeTagStart = text.toLowerCase().indexOf('</style>', startTagEnd + 1);
                if (closeTagStart !== -1) {
                    let styleContent = text.substring(startTagEnd + 1, closeTagStart);
                    let cleanedStyle = stripCssComments(styleContent);
                    out.push(cleanedStyle);
                    i = closeTagStart;
                    continue;
                }
            }
        }

        if (text.substring(i, i + 7).toLowerCase() === '<script') {
            let startTagEnd = text.indexOf('>', i);
            if (startTagEnd !== -1) {
                out.push(text.substring(i, startTagEnd + 1));
                let closeTagStart = text.toLowerCase().indexOf('</script>', startTagEnd + 1);
                if (closeTagStart !== -1) {
                    let scriptContent = text.substring(startTagEnd + 1, closeTagStart);
                    let cleanedScript = stripJsComments(scriptContent);
                    out.push(cleanedScript);
                    i = closeTagStart;
                    continue;
                }
            }
        }

        out.push(text[i]);
        i++;
    }
    return out.join("");
}

function walkDir(dir, callback) {
    const list = fs.readdirSync(dir);
    for (const file of list) {
        const filepath = path.join(dir, file);
        const stat = fs.statSync(filepath);
        if (stat && stat.isDirectory()) {
            // Skip ignored directories
            if (dirsToDelete.includes(file) || file === ".git") {
                continue;
            }
            walkDir(filepath, callback);
        } else {
            callback(filepath, file);
        }
    }
}

function cleanCommentsInProductionFiles() {
    console.log("Removing comments from production files...");
    walkDir(baseDir, (filepath, file) => {
        const ext = path.extname(file).toLowerCase();
        if (![".html", ".css", ".js"].includes(ext)) {
            return;
        }

        if (file === "cleanup_workspace.js" || file === "cleanup_workspace.py") {
            return;
        }

        console.log(`Processing: ${filepath}`);
        try {
            const content = fs.readFileSync(filepath, 'utf8');
            let cleaned = "";
            if (ext === ".html") {
                cleaned = stripHtmlComments(content);
            } else if (ext === ".css") {
                cleaned = stripCssComments(content);
            } else if (ext === ".js") {
                cleaned = stripJsComments(content);
            }

            fs.writeFileSync(filepath, cleaned, 'utf8');
        } catch (err) {
            console.error(`Error processing ${filepath}:`, err.message);
        }
    });
}

// Run tasks
deleteUnnecessary();
cleanCommentsInProductionFiles();
console.log("Done!");

File: cleanup_workspace.py
import os
import shutil
import re

base_dir = r"c:\Users\User\Raizys-site"

files_to_delete = [
    "add-topbar.ps1",
    "byte-clean.ps1",
    "check_active.js",
    "check_all_tags.js",
    "check_boms.js",
    "check_cal_f_rules.js",
    "check_css_encoding.js",
    "check_css_sections.js",
    "check_fixes.ps1",
    "check_home_sections.js",
    "check_null_bytes.js",
    "check_solucoes.js",
    "check_tags.js",
    "clean-artifacts.ps1",
    "clean_artifacts.py",
    "close.js",
    "correct_words.txt",
    "bad_words.txt",
    "css_additions.css",
    "diff.txt",
    "final-clean.ps1",
    "final-surgical-clean.ps1",
    "find_bad_words.ps1",
    "find_flying_cal_styles.js",
    "find_footer_styles.js",
    "fix_all.js",
    "fix_chars.ps1",
    "fix_encoding.js",
    "fix_final.py",
    "fix_forms.js",
    "fix_home.js",
    "fix_mobile_cache.js",
    "git.ignore",
    "inline_css.js",
    "migrate_web3forms.js",
    "precise-clean.ps1",
    "read_agenda_css.js",
    "read_all_changes.js",
    "read_css_diff.js",
    "read_css_sections.js",
    "read_default_eco_bar.js",
    "read_diff.js",
    "read_eco_bar_css.js",
    "read_header_diff.js",
    "read_header_selectors.js",
    "read_home_changes.js",
    "reverse_double_encoding.py",
    "standardize-and-clean.ps1",
    "surgical-clean.ps1",
    "test.txt",
    "update-footer-address.ps1",
    "update-header.ps1",
    "update-pages.ps1",
    "update-version.ps1",
    "update_footer.py",
    ".FullName",
    "assets/1.png",
    "assets/2.png",
    "assets/header-reference.png",
    "assets/hero-reference.png",
    "assets/logo-icon.png",
    "assets/logo-text.png",
    "assets/partner.png",
    "assets/pet_hero_illustration.png"
]

dirs_to_delete = [
    "backup_temp_2",
    "backup_temp_clean",
    "temp_zip"
]

def delete_unnecessary():
    print("Deleting unnecessary files...")
    for f in files_to_delete:
        path = os.path.join(base_dir, f)
        if os.path.exists(path):
            try:
                os.remove(path)
                print(f"Deleted file: {f}")
            except Exception as e:
                print(f"Error deleting file {f}: {e}")
                
    print("Deleting unnecessary directories...")
    for d in dirs_to_delete:
        path = os.path.join(base_dir, d)
        if os.path.exists(path) and os.path.isdir(path):
            try:
                shutil.rmtree(path)
                print(f"Deleted directory: {d}")
            except Exception as e:
                print(f"Error deleting directory {d}: {e}")

def strip_css_comments(text):
    out = []
    i = 0
    n = len(text)
    in_comment = False
    in_string_single = False
    in_string_double = False
    while i < n:
        c = text[i]
        next_c = text[i+1] if i + 1 < n else ""
        if not in_comment:
            if not in_string_single and not in_string_double:
                if c == '/' and next_c == '*':
                    in_comment = True
                    i += 1
                elif c == "'":
                    in_string_single = True
                    out.append(c)
                elif c == '"':
                    in_string_double = True
                    out.append(c)
                else:
                    out.append(c)
            elif in_string_single:
                out.append(c)
                if c == '\\':
                    out.append(next_c)
                    i += 1
                elif c == "'":
                    in_string_single = False
            elif in_string_double:
                out.append(c)
                if c == '\\':
                    out.append(next_c)
                    i += 1
                elif c == '"':
                    in_string_double = False
        else:
            if c == '*' and next_c == '/':
                in_comment = False
                i += 1
        i += 1
    return "".join(out)

def strip_js_comments(text):
    out = []
    i = 0
    n = len(text)
    state = "NORMAL"
    
    def get_last_non_ws():
        for c in reversed(out):
            if not c.isspace():
                return c
        return ""

    while i < n:
        c = text[i]
        next_c = text[i+1] if i + 1 < n else ""
        
        if state == "NORMAL":
            if c == '"':
                state = "STR_DOUBLE"
                out.append(c)
            elif c == "'":
                state = "STR_SINGLE"
                out.append(c)
            elif c == '`':
                state = "STR_TEMPLATE"
                out.append(c)
            elif c == '/' and next_c == '/':
                state = "LINE_COMMENT"
                i += 1
            elif c == '/' and next_c == '*':
                state = "BLOCK_COMMENT"
                i += 1
            elif c == '/':
                last_char = get_last_non_ws()
                if last_char and (last_char.isalnum() or last_char in ")$_]"):
                    out.append(c)
                else:
                    state = "REGEX"
                    out.append(c)
            else:
                out.append(c)
                
        elif state == "STR_DOUBLE":
            out.append(c)
            if c == '\\':
                out.append(next_c)
                i += 1
            elif c == '"':
                state = "NORMAL"
                
        elif state == "STR_SINGLE":
            out.append(c)
            if c == '\\':
                out.append(next_c)
                i += 1
            elif c == "'":
                state = "NORMAL"
                
        elif state == "STR_TEMPLATE":
            out.append(c)
            if c == '\\':
                out.append(next_c)
                i += 1
            elif c == '`':
                state = "NORMAL"
                
        elif state == "REGEX":
            out.append(c)
            if c == '\\':
                out.append(next_c)
                i += 1
            elif c == '[':
                state = "REGEX_CLASS"
            elif c == '/':
                state = "NORMAL"
                
        elif state == "REGEX_CLASS":
            out.append(c)
            if c == '\\':
                out.append(next_c)
                i += 1
            elif c == ']':
                state = "REGEX"
                
        elif state == "LINE_COMMENT":
            if c == '\n':
                state = "NORMAL"
                out.append(c)
                
        elif state == "BLOCK_COMMENT":
            if c == '*' and next_c == '/':
                state = "NORMAL"
                i += 1
                
        i += 1
        
    return "".join(out)

def strip_html_comments(text):
    out = []
    i = 0
    n = len(text)
    
    while i < n:
        if text[i:i+4] == '<!--':
            i += 4
            while i < n and text[i:i+3] != '-->':
                i += 1
            i += 3
            continue
            
        if text[i:i+6].lower() == '<style':
            start_tag_end = text.find('>', i)
            if start_tag_end != -1:
                out.append(text[i:start_tag_end+1])
                close_tag_start = text.lower().find('</style>', start_tag_end+1)
                if close_tag_start != -1:
                    style_content = text[start_tag_end+1:close_tag_start]
                    cleaned_style = strip_css_comments(style_content)
                    out.append(cleaned_style)
                    i = close_tag_start
                    continue
                    
        if text[i:i+7].lower() == '<script':
            start_tag_end = text.find('>', i)
            if start_tag_end != -1:
                out.append(text[i:start_tag_end+1])
                close_tag_start = text.lower().find('</script>', start_tag_end+1)
                if close_tag_start != -1:
                    script_content = text[start_tag_end+1:close_tag_start]
                    cleaned_script = strip_js_comments(script_content)
                    out.append(cleaned_script)
                    i = close_tag_start
                    continue
                    
        out.append(text[i])
        i += 1
        
    return "".join(out)

def clean_comments_in_production_files():
    print("Removing comments from production files...")
    for root, _, files in os.walk(base_dir):
        # Skip temporary directories or ignored directories
        if any(ignored in root for ignored in dirs_to_delete + [".git"]):
            continue
            
        for file in files:
            ext = os.path.splitext(file)[1].lower()
            if ext not in [".html", ".css", ".js"]:
                continue
                
            # Skip the cleanup script itself
            if file == "cleanup_workspace.py":
                continue
                
            filepath = os.path.join(root, file)
            print(f"Processing: {filepath}")
            
            try:
                # Read using utf-8-sig to handle BOM or utf-8 safely
                with open(filepath, 'r', encoding='utf-8-sig', errors='ignore') as f:
                    content = f.read()
                
                if ext == ".html":
                    cleaned = strip_html_comments(content)
                elif ext == ".css":
                    cleaned = strip_css_comments(content)
                elif ext == ".js":
                    cleaned = strip_js_comments(content)
                else:
                    continue
                
                with open(filepath, 'w', encoding='utf-8', newline='') as f:
                    f.write(cleaned)
                    
            except Exception as e:
                print(f"Error processing {filepath}: {e}")

if __name__ == "__main__":
    delete_unnecessary()
    clean_comments_in_production_files()
    print("Done!")
